dns-server/src/dns_server.rs

```diff
@@ -284,30 +335,38 @@ async fn handle_dns_message(
                 (RecordType::A, DnsRecord::A(_)) => true,
                 (RecordType::AAAA, DnsRecord::Aaaa(_)) => true,
                 (RecordType::SRV, DnsRecord::Srv(_)) => true,
+                (RecordType::NS, DnsRecord::Ns(_)) => true,
+                (RecordType::SOA, DnsRecord::Soa(_)) => true,
```

Comment:
one interesting detail here: RFC 1034 describes negative caching as returning an SOA in an authoritative name error, where the `minimum` field of the SOA is used as the TTL for the negative result. that's probably one of the more common ways you'd run into an SOA record if you're not operating nameservers yourself. this only returns an SOA record if you've queried about an SOA record.

in the negative caching case we'd want to return this SOA record in the `soa` list provided to `MessageResponseBuilder`. somewhat different kind of plumbing, but it'd be relatively straightforward after this.